_posts/2019-06-06-NeuralODEs.markdown

```diff
@@ -0,0 +1,388 @@
+
+  **Topics**:
+
+  1. ResNets
+  2. ResNets and ODEs
+
+  **Required Reading**:
+
+  1. ResNets: 
+     * [ResNets](https://www.coursera.org/lecture/convolutional-neural-networks/resnets-HAhz9) 
+     * [An Overview of ResNet and its Variants](https://towardsdatascience.com/an-overview-of-resnet-and-its-variants-5281e2f56035)
+  2. ResNets and ODEs: 
+     * Sections 1-3 from [Multi-level Residual Networks from Dynamical Systems View](https://arxiv.org/pdf/1710.10348.pdf)
+     * [Reversible Architectures for Arbitrarily Deep Residual Neural Networks](https://arxiv.org/abs/1709.03698)
+     * Invertible ResNets: [The Reversible Residual Network: Backpropagation Without Storing Activations](https://arxiv.org/pdf/1707.04585.pdf)
+     * [Stable Architectures for Deep Neural Networks](https://arxiv.org/pdf/1705.03341.pdf)
+     
+  **Notes**: Here is a [link](/assets/nodes_notes/week3.pdf) to our notes for the lesson.
+  
+  **Optional Reading**:
+
+  1. The original ResNets paper: [Deep Residual Learning for Image Recognition](https://arxiv.org/abs/1512.03385)
+  2. Another blog post on ResNets: [Understanding and Implementing Architectures of ResNet and ResNeXt for state-of-the-art Image Classification](https://medium.com/@14prakash/understanding-and-implementing-architectures-of-resnet-and-resnext-for-state-of-the-art-image-cf51669e1624)
+  
+  **Questions**:
+
+  1. Do you understand why adding ‘residual layers’ should not degrade the network performance?
+     <details><summary>Solution</summary>
+     <p>
+     See the notes from this week.
+     </p>
+     </details>
+
+  2. How do the authors of (Multi-level Residual Networks from Dynamical Systems View) explain the phenomena of still having almost as good performances in residual networks when removing a layer?
+     <details><summary>Solution</summary>
+     <p>
+     See section 3.3 in the paper.
+     </p>
+     </details>
+
+  3. Implement your favourite ResNet variant 
+
+<br />
+
+# 4 Normalising Flows
+  **Motivation**: In this class we take a little detour through the topic of Normalising Flows. This is used for density estimation and generative modeling, and it is another model which can be seen a time-discretisation of its continuous-time counterpart. Understanding it is necessary to understand continuous normalizing flows, one of the applications of neural ODEs.
+
+  **Topics**:
+
+  1. Normalising Flows
+  2. End-to-end implementations with neural nets
+
+  **Required Reading**:
+
+  1. [Density Estimation by Dual Ascent of the Log-likelihood](https://math.nyu.edu/faculty/tabak/publications/CMSV8-1-10.pdf) (you can skip Section 3) (DE)
```

Comment:
"(you can skip Section 3)" --> "(Skip Section 3)"

Reply:
Move this to the top with the other ones that you abbreviate.

Reply:
✅